### PR TITLE
RyuJIT - throughput improvements, PInvoke transitions tweaks:

### PR DESCRIPTION
**Pull request by jkotas**
- Minor compilation throughput improvements:
     - Avoid redundant calls to getMethodAttribs in impMarkInlineCandidate by reusing the value from CALL_INFO
     - Avoid redundant calls to getFunctionEntryPoint in codegen by reusing the value computed during lower
     - Stop checking CORINFO_FLG_NATIVE flag for inliner observations. It is dead flag, not set anywhere
- Tweaks for helper-based PInvoke transitions (used CoreRT only today):
     - Stop emitting CORINFO_HELP_INIT_PINVOKE_FRAME. Just having the two simple begin/end helpers is better.
     - Stop passing size of arguments in secret register.
     - Enable direct calls for PInvoke targets

### CODE REVIEW

**Comment by jkotas** [comment]
@dotnet/jit-contrib PTLA

File: src/jit/gentree.h [hunk 1]
```diff
@@ -2887,6 +2887,7 @@ struct GenTreeCall final : public GenTree
         InlineCandidateInfo * gtInlineCandidateInfo;
         void *  gtStubCallStubAddr;             // GTF_CALL_VIRT_STUB - these are never inlined                
         CORINFO_GENERIC_HANDLE compileTimeHelperArgumentHandle; // Used to track type handle argument of dynamic helpers
+        void * gtDirectCallAddress; // Used to pass direct call address between lower and codegen
```
Comment (sivarv) [inline src/jit/gentree.h]: (Minor) as per Jit CC, T* instead of T *

e.g. 
void* gtDirectCallAddress;

Comment (sivarv) [inline src/jit/gentree.h]: I would suggest to initialize this field to nullptr in gtNewCallNode() and gtNewIndCallNode().

Further assert in codegenxarch.cpp that gtDirrectCallAddress is not nullptr.

Comment (sivarv) [inline src/jit/gentree.h]: gtCloneExpr() - while cloning a call node this field too needs to be copied.

File: src/jit/importer.cpp [hunk 2]
```diff
@@ -16459,7 +16459,9 @@ BOOL                Compiler::impInlineIsGuaranteedThisDerefBeforeAnySideEffects
 // otherwise build tree context) so when we do the inlining pass we
 // can capture these reasons
 
-void          Compiler::impMarkInlineCandidate(GenTreePtr callNode, CORINFO_CONTEXT_HANDLE exactContextHnd)
+void          Compiler::impMarkInlineCandidate(GenTreePtr callNode, 
+                                               CORINFO_CONTEXT_HANDLE exactContextHnd, 
+                                               CORINFO_CALL_INFO * callInfo)
```
Comment (sivarv) [inline src/jit/importer.cpp]: As per JIT CC use

CORINFO_CALL_INFO* callInfo

**Comment by sivarv** [comment]
Looks good (modulo comments).

File: src/jit/gentree.h (hunk 1 above)
Comment (jkotas) [inline src/jit/gentree.h]: This field is union with `gtInlineCandidateInfo` that is set to null and cloned already - the assumption is that `gtInlineCandidateInfo` is not going to be used after lowering.

**Comment by jkotas** [comment]
@dotnet-bot test Windows_NT x86 ryujit Checked Build and Test   please

**Comment by jkotas** [comment]
Opened #4844 on the intermittent RyuJIT x86 test failure - other PRs are hitting it as well.

**Pull request merged**